transcendence-app/src/chat/infrastructure/db/postgres/chatmember.postgres.repository.ts

```diff
@@ -13,4 +18,31 @@ export class ChatMemberPostgresRepository
   constructor(protected pool: PostgresPool) {
     super(pool, table.CHATMEMBERS);
   }
+
+  async retrieveChatRoomMembers(
+    chatRoomId: string,
+  ): Promise<ChatMemberWithUser[] | null> {
+    const users = await makeQuery<ChatMemberWithUserEntity>(this.pool, {
+      text: `SELECT u."username",
+                    u."avatarId",
+                    u."avatarX",
+                    u."avatarY",
+                    c."ownerId",
+                    cm."admin",
+                    cm."muted",
+                    cm."banned"
+             FROM ${table.USERS} u
+                    INNER JOIN ${this.table} cm
+                               ON cm."userId" = u."id"
+                    LEFT JOIN ${table.CHATS} c
+                              ON c."id" = cm."chatId"
+                                AND u."id" = c."ownerId"
+             WHERE cm."chatId" = $1
+               AND cm."joinedAt" IS NOT NULL`,
```

Comment:
maybe we would like to have this logic in the chatmember.service instead of here, by injecting the IChatPGRepo into it as well (the way we do with the blocks in user), but since we are already joining two tables here, why shouldn't we join three.